test/test-DOMFetcher.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const test = require("sdk/test");
+const tabs = require("sdk/tabs");
+const httpd = require("./lib/httpd");
+const {before, after} = require("sdk/test/utils");
+const {DOMFetcher} = require("lib/DOMFetcher");
+const {Cu} = require("chrome");
+const {doGetFile} = require("./lib/utils");
+
+const {TextDecoder, OS} = Cu.import("resource://gre/modules/osfile.jsm", {});
+const decoder = new TextDecoder();
+
+const PORT = 8099;
+const url = `http://localhost:${PORT}/dummy.html`;
+const otherUrl = `http://localhost:${PORT}/other-dummy.html`;
+
+let srv;
```

Comment:
nit: globals should be in all-caps or prefaced with a `k`. e.g.:

`kSrv` or `SRV`